pkg/component/nodelocaldns/nodelocaldns.go

```diff
@@ -583,34 +581,14 @@ func (c *nodeLocalDNS) containerArg() string {
 }
 
 func (c *nodeLocalDNS) forceTcpToClusterDNS() string {
-	fromSpec := true
-	if c.values.Config != nil && c.values.Config.ForceTCPToClusterDNS != nil {
-		fromSpec = *c.values.Config.ForceTCPToClusterDNS
-	}
-
-	fromAnnotation := true
-	if annotationValue, err := strconv.ParseBool(c.values.ShootAnnotations[v1beta1constants.AnnotationNodeLocalDNSForceTcpToClusterDns]); err == nil {
-		fromAnnotation = annotationValue
-	}
-
-	if fromAnnotation && fromSpec {
+	if c.values.Config != nil && c.values.Config.ForceTCPToClusterDNS != nil && *c.values.Config.ForceTCPToClusterDNS {
```

Comment:
Please revert the logic here. We should not change the default, i.e. without specifying the option it should force tcp.